### PR TITLE
Update Gradle to 7.4

### PR DESCRIPTION
**Pull request by adrianclay**
## What

Gradle update to 7.4

Also remove some unused gradle wrapper code from root of repo.

## Why

Current gradle versions don't support Java 17. Java 17 is the next Long Term Support version of Java. Updating Gradle removes a barrier to updating java.

## Type of change

- [x] Internal change (non-breaking change with no effect on the functionality affecting end users)

## Checklist:

- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by adrianclay** [comment]
Superceeds #564

**Comment by ole4ryb** [comment]
in ..\smoke-tests\gradle\wrapper\gradle-wrapper.properties we use ver. 7.4.2. Maybe we can upgrade everything to ver. 7.4 to it is consistent across the whole application ?

File: e2e-tests/gradlew [hunk 1]
```diff
@@ -17,67 +17,101 @@
...
-SAVED="`pwd`"
-cd "`dirname \"$PRG\"`/" >/dev/null
-APP_HOME="`pwd -P`"
-cd "$SAVED" >/dev/null
+
+APP_HOME=$( cd "${APP_HOME:-./}" && pwd -P ) || exit
 
 APP_NAME="Gradle"
-APP_BASE_NAME=`basename "$0"`
+APP_BASE_NAME=${0##*/}
```
Comment (ole4ryb) [inline e2e-tests/gradlew:86]: why are we replacing `basename "$0"` with ${0##*/} ?

Comment (adrianclay) [reply e2e-tests/gradlew:86]: Sounds like a question for the Gradle wrapper maintainers, given this file is their copyright, but you can find [more information on that syntax here](https://unix.stackexchange.com/questions/214465/what-does-prog-0-mean-in-a-bash-script).

**Review by ole4ryb — APPROVED** [review]

**Pull request merged**